src/plugins/dashboards_security/server/auth/types/saml/routes.ts

```diff
@@ -0,0 +1,290 @@
+/*
+ *   Copyright OpenSearch Contributors
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import { schema } from '@osd/config-schema';
+import {
+  IRouter,
+  SessionStorageFactory,
+  OpenSearchDashboardsRequest,
+} from '../../../../../../../src/core/server';
+import { SecuritySessionCookie } from '../../../session/security_cookie';
+import { SecurityPluginConfigType } from '../../..';
+import { SecurityClient } from '../../../backend/opensearch_security_client';
+import { CoreSetup } from '../../../../../../../src/core/server';
+import { validateNextUrl } from '../../../utils/next_url';
+import { AuthType, SAML_AUTH_LOGIN, SAML_AUTH_LOGOUT } from '../../../../common';
+import { compileSchema } from 'ajv/dist/compile';
+import { XMLParser } from "fast-xml-parser";
+import { AuthToken } from './utils/AuthToken';
+
+export class SamlAuthRoutes {
+  constructor(
+    private readonly router: IRouter,
+    // @ts-ignore: unused variable
+    private readonly config: SecurityPluginConfigType,
+    private readonly sessionStorageFactory: SessionStorageFactory<SecuritySessionCookie>,
+    private readonly securityClient: SecurityClient,
+    private readonly coreSetup: CoreSetup
+  ) {}
+
+  public setupRoutes() {
+    this.router.get(
+      {
+        path: SAML_AUTH_LOGIN,
+        validate: {
+          query: schema.object({
+            nextUrl: schema.maybe(
+              schema.string({
+                validate: validateNextUrl,
+              })
+            ),
+          }),
+        },
+        options: {
+          authRequired: false,
+        },
+      },
+      async (context, request, response) => {
+        if (request.auth.isAuthenticated) {
+          return response.redirected({
+            headers: {
+              location: `${this.coreSetup.http.basePath.serverBasePath}/app/opensearch-dashboards`,
+            },
+          });
+        }
+
+        try {
+          const samlHeader = await this.securityClient.getSamlHeader(request);
```

Comment:
This securityClient

Reply:
The response is hardcode. It doesn't call backend API. https://github.com/elastic-analytics/OpenSearch-Dashboards/pull/41/files#diff-ec3bb94520a80a2438569c440d593c9023b156fc932335e7e68de3cfbb0e167cR192-R195